sdks/python/apache_beam/runners/direct/direct_runner_test.py

```diff
@@ -52,6 +54,15 @@ def test_waiting_on_result_stops_executor_threads(self):
 
       post_test_threads = set(t.ident for t in threading.enumerate())
       new_threads = post_test_threads - pre_test_threads
+      if len(new_threads) > 0:
+        for i in range(1,6):
+          logging.info('%s: Wait %s second(s) for thread(s) to close.' %
+                       (runner, i))
+          time.sleep(i)
+          post_test_threads = set(t.ident for t in threading.enumerate())
+          new_threads = post_test_threads - pre_test_threads
+          if len(new_threads) == 0:
+            break
```

Comment:
When I test at local, 10% of time (sample size: 4,000) threads are closed after waiting 1 second with `DirectRunner` and `SwitchingDirectRunner`. 90% of time threads are closed immediately with all runners.